src/ArrayExpressionBuilder.php

```diff
@@ -67,23 +75,33 @@ public function build(ExpressionInterface $expression, array &$params = []): str
      */
     protected function buildPlaceholders(ExpressionInterface $expression, array &$params): array
     {
-        /** @var ArrayExpression $expression */
+        $placeholders = [];
+
+        /**
+         *  @var ArrayExpression $expression
+         *  @var mixed $value
+         */
         $value = $expression->getValue();
 
-        $placeholders = [];
-        if ($value === null || (!is_array($value) && !$value instanceof Traversable)) {
+        if ($value === null || !is_array($value) && !$value instanceof Traversable) {
```

Comment:
```suggestion
        if (!is_array($value) && !$value instanceof Traversable) {
```

You was initially right about the change...